### PR TITLE
[SPARK-32402][SQL][FOLLOW-UP] Add case sensitivity tests for column resolution in ALTER TABLE

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
Add case sensitivity tests for column resolution in ALTER TABLE

### Why are the changes needed?
To make sure `spark.sql.caseSensitive` works for `ResolveAlterTableChanges`

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
new test

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTableCatalogSuite.scala [hunk 1]
```diff
@@ -318,4 +319,69 @@ class JDBCTableCatalogSuite extends QueryTest with SharedSparkSession {
...
+      withSQLConf(SQLConf.CASE_SENSITIVE.key -> "false") {
+        sql("ALTER TABLE h2.test.alt_table DROP COLUMN C3")
+        expectedSchema = new StructType().add("c1", IntegerType)
+        t = spark.table("h2.test.alt_table")
+        assert(t.schema === expectedSchema)
+      }
+
+      withSQLConf(SQLConf.CASE_SENSITIVE.key -> "true") {
+        val msg = intercept[AnalysisException] {
+        sql("ALTER TABLE h2.test.alt_table ALTER COLUMN C1 TYPE DOUBLE")
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTableCatalogSuite.scala:360]: indentation?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTableCatalogSuite.scala [hunk 2]
```diff
@@ -318,4 +319,69 @@ class JDBCTableCatalogSuite extends QueryTest with SharedSparkSession {
...
+      withSQLConf(SQLConf.CASE_SENSITIVE.key -> "false") {
+        sql("ALTER TABLE h2.test.alt_table ALTER COLUMN C1 TYPE DOUBLE")
+        expectedSchema = new StructType().add("c1", DoubleType)
+        t = spark.table("h2.test.alt_table")
+        assert(t.schema === expectedSchema)
+      }
+
+      withSQLConf(SQLConf.CASE_SENSITIVE.key -> "true") {
+        val msg = intercept[AnalysisException] {
+        sql("ALTER TABLE h2.test.alt_table ALTER COLUMN C1 DROP NOT NULL")
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTableCatalogSuite.scala:374]: indentation?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTableCatalogSuite.scala [hunk 3]
```diff
@@ -318,4 +319,69 @@ class JDBCTableCatalogSuite extends QueryTest with SharedSparkSession {
       assert(msg.contains("Table not found"))
     }
   }
+
+  test("ALTER TABLE case sensitivity") {
+    withTable("h2.test.alt_table") {
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/v2/jdbc/JDBCTableCatalogSuite.scala:324]: Although I understand this is the patter in this suite, let's declare a variable for this string and reuse it in this new test case, @huaxingao .

**Comment by huaxingao** [comment]
@dongjoon-hyun Thanks for the comments! I have updated the code. Please take another check.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM (Pending CIs).
Thank you, @huaxingao !

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by huaxingao** [comment]
Thanks! @dongjoon-hyun @HyukjinKwon 
FYI @cloud-fan

**Pull request closed without merging**